main/test/ca/uwaterloo/flix/tools/TestPackager.scala

```diff
@@ -40,6 +46,48 @@ class TestPackager extends FunSuite {
     assert(jarPath.getFileName.toString.startsWith(ProjectPrefix))
   }
 
+  test("build-jar generates ZIP entries with fixed time") {
+    val p = Files.createTempDirectory(ProjectPrefix)
+    Packager.init(p, DefaultOptions)
+    Packager.buildJar(p, DefaultOptions)
+
+    val packageName = p.getFileName.toString
+    val packagePath = p.resolve(packageName + ".jar")
+    val format = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss")
+    for (e <- new ZipFile(packagePath.toFile).entries().asScala) {
+      val time = new Date(e.getTime)
+      val formatted = format.format(time)
+      assert(formatted.equals("2014-06-27 00:00:00"))
+    }
+  }
+
+  def calcHash(p: Path): String = {
+    val buffer = new Array[Byte](8192)
+    val sha = MessageDigest.getInstance("SHA-256")
+    Using(new DigestInputStream(Files.newInputStream(p), sha)) { input =>
+      while (input.read(buffer) != -1) { }
+      sha.digest.map("%02x".format(_)).mkString
+    }.get
+  }
+
+  test("build-jar always generates package that is byte-for-byte exactly the same") {
+    val p = Files.createTempDirectory(ProjectPrefix)
+    Packager.init(p, DefaultOptions)
+    val packageName = p.getFileName.toString
+    val packagePath = p.resolve(packageName + ".jar")
+
+    Packager.buildJar(p, DefaultOptions)
+    def hash1 = calcHash(packagePath)
+
+    Files.delete(packagePath)
```

Comment:
Flix has a very strict policy on never deleting files.  (Out of an abundance of caution).

Just remove this line. We create a temp. dir. which will be cleaned by the OS anyway.